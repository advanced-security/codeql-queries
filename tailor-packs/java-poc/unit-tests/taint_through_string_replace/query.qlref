Security/CWE/CWE-079/XSS.ql
