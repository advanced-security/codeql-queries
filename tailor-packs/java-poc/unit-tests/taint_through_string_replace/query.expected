edges
| Test.java:10:18:10:45 | getParameter(...) : String | Test.java:11:11:11:15 | taint : String |
| Test.java:11:11:11:15 | taint : String | Test.java:11:11:11:48 | replace(...) : String |
| Test.java:11:11:11:48 | replace(...) : String | Test.java:12:30:12:41 | ... + ... |
nodes
| Test.java:10:18:10:45 | getParameter(...) : String | semmle.label | getParameter(...) : String |
| Test.java:11:11:11:15 | taint : String | semmle.label | taint : String |
| Test.java:11:11:11:48 | replace(...) : String | semmle.label | replace(...) : String |
| Test.java:12:30:12:41 | ... + ... | semmle.label | ... + ... |
subpaths
#select
| Test.java:12:30:12:41 | ... + ... | Test.java:10:18:10:45 | getParameter(...) : String | Test.java:12:30:12:41 | ... + ... | Cross-site scripting vulnerability due to $@. | Test.java:10:18:10:45 | getParameter(...) | user-provided value |
